### ISSUE TITLE
STATUS does not seem to work, never gets into the journal?

### ISSUE THREAD

**Issue by dachshund-digital** [bb4242/sdnotify#4]
May 17 02:15:47 [redacted] python[1558]: [50B blob data]
May 17 02:15:47 [redacted] python[1558]: [52B blob data]
May 17 02:15:47 [redacted] python[1558]: [53B blob data]

Code...
import sdnotify
theNotification=sdnotify.SystemdNotifier()
theNotification.notify('STATUS={0}'.format(theAccumulator))

theAccumulator is just a counting int var.  Start and Stop methods seem to work fine.

**Comment by dachshund-digital** [comment]
bump... no one actively supporting this anymore?

**Comment by bb4242** [comment]
STATUS information shows up in the output of `systemctl status`:
```
$ systemctl --user status test
● test.service - A test service written in Python
   Loaded: loaded (/home/bbethke/.config/systemd/user/test.service; static; vendor preset: enabled)
   Active: active (running) since Sun 2019-06-09 15:30:30 PDT; 1min 8s ago
 Main PID: 10463 (python)
   Status: "Count is 35"
   CGroup: /user.slice/user-1000.slice/user@1000.service/test.service
           └─10463 /usr/bin/python /home/bbethke/test.py

Jun 09 15:31:20 roland python[10463]: Running... 26
Jun 09 15:31:22 roland python[10463]: Running... 27
Jun 09 15:31:24 roland python[10463]: Running... 28
Jun 09 15:31:26 roland python[10463]: Running... 29
Jun 09 15:31:28 roland python[10463]: Running... 30
Jun 09 15:31:30 roland python[10463]: Running... 31
Jun 09 15:31:32 roland python[10463]: Running... 32
Jun 09 15:31:34 roland python[10463]: Running... 33
Jun 09 15:31:36 roland python[10463]: Running... 34
Jun 09 15:31:38 roland python[10463]: Running... 35
```
As far as I know, systemd does not log status information to the journal by design.

**Comment by bb4242** [comment]
PS - If you want to output something to the journal, it sufficient to simply call `print(...)` since systemd will capture stdout from the process and log it.

**Comment by dachshund-digital** [comment]
In the standard sdnotify list there is an explicit status option.  READY, i.e. READY=1 for start works.  STOPPING for stop works.  But when I try to send STATUS to sdnotify via your code, all I get is BLOB references in the journal trace, i.e. service logging.  Per the sdnotify design specific it should pass the string provided as STATUS='string' to the service journal, i.e. logging, when systemctl status <service> is invoked.  This just like the code you would write for start and stop messages via sdnotify.

**Comment by bb4242** [comment]
Hmm, yeah, those BLOB references in the journal seem strange.  Have you tried running the example service in the [readme](https://github.com/bb4242/sdnotify/blob/master/README.md)?  I can confirm that the STATUS information works fine for me using that code.  I tested using both python2 and python3 on Ubuntu 18.04 (which uses systemd 237).  I think Debian 9 uses [systemd 232](https://packages.debian.org/stretch/systemd).  Maybe there is a bug in that version of systemd.  Can you provide a minimal complete example of your code that is generating the BLOB references?

**Comment by dachshund-digital** [comment]
Actually I will do one better... I will create a virtual machine using straight Debian 9 stretch, rather than Raspbian, and test there, then change systemd version, etc. and test again, etc.  I want to make sure it is not something unique to ARM architecture implementation versus Intel/AMD as well.  Would be surprised if so... but stranger things have happened.  Unfortunately the current code is part of a rather large python project that has its own framework, inherited classes, etc. so not easy to parse it down.  Hence the use of the VM, where I can just duplicate the entire project and test.

**Comment by bb4242** [comment]
OK, those sound like they'll be useful tests. :)  Let me know what you find out!

**Comment by dachshund-digital** [comment]
Still need to test... But I doubt I need to do more testing now... I found the following that seems to explain what is going on...

"... journalctl tries to detect if some journal data is binary rather than text, and if so, it shows the "blob data" output instead of the message. Its heuristic for telling the difference seems a little buggy, and hides things that could be shown sometimes. The "-a" flag disables this (and would then print actual binary data as well if any ever appeared)."

Reference... https://bbs.archlinux.org/viewtopic.php?id=149617

And the -a trick does expose the text that is not being parsed correctly.  I realized that when I send (unintended) an escape sequence to the journal that is one example of when the blob results.  For example...  with journalctl -a -xe...

Jun 10 21:55:03 [redacted] systemd[1]: Started Event Listener Service.
-- Subject: Unit EventListener.service has finished start-up
-- Defined-By: systemd
-- Support: https://www.debian.org/support
--
-- Unit EventListener.service has finished starting up.
--
-- The start-up result is done.
Jun 10 21:55:04 [redacted] EventListener[4553]: [2019-06-10 21:55:04 GMT]  Event Listener * 1.0.0 * 06/09/2019
Jun 10 21:55:04 [redacted] EventListener[4553]: [2019-06-10 21:55:04 GMT]  Event Listener...
Jun 10 21:55:04 [redacted] EventListener[4553]: [2019-06-10 21:55:04 GMT]  Start...
Jun 10 21:55:04 [redacted] EventListener[4553]: [2019-06-10 21:55:04 GMT]  ...Start.
Jun 10 21:55:04 [redacted] EventListener[4553]: [2019-06-10 21:55:04 GMT]  Status...
Jun 10 21:55:04 [redacted] EventListener[4553]: [2019-06-10 21:55:04 GMT]  ...Status.
Jun 10 21:55:05 [redacted] EventListener[4553]: [2019-06-10 21:55:05 GMT]  Status...
Jun 10 21:55:16 [redacted] EventListener[4553]: [2019-06-10 21:55:16 GMT]  ...Status.
Jun 10 21:55:16 [redacted] systemd[1]: Stopping Event Listener Service...
-- Subject: Unit EventListener.service has begun shutting down
-- Defined-By: systemd
-- Support: https://www.debian.org/support
--
-- Unit EventListener.service has begun shutting down.
Jun 10 21:55:16 [redacted] EventListener[4553]:
Jun 10 21:55:16 [redacted] EventListener[4553]: [2019-06-10 21:55:16 GMT]  Control Break (Control-C Interrupt)? SIGINT
Jun 10 21:55:16 [redacted] EventListener[4553]: [2019-06-10 21:55:16 GMT]  Interrupt (SIGINT)...
Jun 10 21:55:16 [redacted] EventListener[4553]: [2019-06-10 21:55:16 GMT]  ...Interrupt (SIGINT).
Jun 10 21:55:16 [redacted] EventListener[4553]: [2019-06-10 21:55:16 GMT]  Stop...
Jun 10 21:55:16 [redacted] EventListener[4553]: [2019-06-10 21:55:16 GMT]  ...Stop.
Jun 10 21:55:16 [redacted] EventListener[4553]: [2019-06-10 21:55:16 GMT]  ...Event Listener.
Jun 10 21:55:16 [redacted] systemd[1]: Stopped Event Listener Service.
-- Subject: Unit EventListener.service has finished shutting down
-- Defined-By: systemd
-- Support: https://www.debian.org/support
--
-- Unit EventListener.service has finished shutting down.

Where as without the -a the Control Break, Interrupt SIGINT related text shows up as a blob... because it is an ansi esc sequence (colorized) text line(s).  Easy enough to avoid in code design, now that I understand what the heck was going on!  

We use colorized text to highlight various warnings and events in the console text output... sort of like how dmesg does things, so sending SIGINT to a service is one such.  Given during development we are basically echoing almost all output to journal, the ansi escape sequenced messages, i.e. text gets dumped in as well.  Just need to write a little code for production, that will either strip the ansi esc sequence, or never post such messages to journal.  What is kind of dumb is that journal is not smart enough to understand basic ansi escape sequences?  That standard has been around for what... the 1970s?  Actually started development in 1976 and 1979 accepted standard per Wikipedia... so 40 or so years ago?  LOL.
